packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,412 @@
+
+  /// Allows changing the style of the text.
+  ///
+  /// Defaults to the gray [CupertinoColors.secondaryLabel] iOS color.
+  final TextStyle? style;
+
+  /// A hint placeholder text that appears when the text entry is empty.
+  ///
+  /// Defaults to 'Search' localized in each supported language.
+  final String? placeholder;
+
+  /// Sets the style of the placeholder of the textfield.
+  ///
+  /// Defaults to the gray [CupertinoColors.secondaryLabel] iOS color.
+  final TextStyle? placeholderStyle;
+
+  /// Sets the decoration for the text field.
+  ///
+  /// This property is automatically set using the [backgroundColor] and
+  /// [borderRadius] properties, which both have default values. Therefore,
+  /// [decoration] has a default value upon building the widget. It is designed
+  /// to mimic the look of a `UISearchTextField`.
+  final BoxDecoration? decoration;
+
+  /// Set the [decoration] property's background color.
+  ///
+  /// Can't be set along with the [decorator]. Defaults to the translucent
+  /// [CupertinoColors.tertiarySystemFill] iOS color.
+  final Color? backgroundColor;
+
+  /// Sets the [decoration] property's border radius.
+  ///
+  /// Can't be set along with the [decorator]. Defaults to 9 px circular
+  /// corner radius.
+  // TODO(DanielEdrisian): Must make border radius continuous, see
+  // https://github.com/flutter/flutter/issues/13914.
+  final BorderRadius? borderRadius;
+
+  /// Sets the padding insets for the text and placeholder.
+  ///
+  /// Cannot be null. Defaults to padding that replicates the
+  /// `UISearchTextField` look. The inset values were determined using the
+  /// comparison tool in https://github.com/flutter/platform_tests/.
+  final EdgeInsetsGeometry padding;
+
+  /// Sets the color for the suffix and prefix icons.
+  ///
+  /// Cannot be null. Defaults to [CupertinoColors.secondaryLabel].
+  final Color itemColor;
+
+  /// Sets the base icon size for the suffix and prefix icons.
+  ///
+  /// Cannot be null. The size of the icon is scaled using the accessibility
+  /// font scale settings. Defaults to [20.0].
+  final double itemSize;
+
+  /// Sets the padding insets for the suffix.
+  ///
+  /// Cannot be null. Defaults to padding that replicates the
+  /// `UISearchTextField` suffix look. The inset values were determined using
+  /// the comparison tool in https://github.com/flutter/platform_tests/.
+  final EdgeInsetsGeometry prefixInsets;
+
+  /// Sets the padding insets for the prefix.
+  ///
+  /// Cannot be null. Defaults to padding that replicates the
+  /// `UISearchTextField` prefix look. The inset values were determined using
+  /// the comparison tool in https://github.com/flutter/platform_tests/.
+  final EdgeInsetsGeometry suffixInsets;
+
+  /// Sets the suffix widget's icon.
+  ///
+  /// Cannot be null. Defaults to the X-Mark [CupertinoIcons.xmark_circle_fill].
+  /// The suffix is customizable so that users can override it with other
```

Comment:
nit: this API documentation is meant for developers, "users" from their perspective usually refers to end-users. Maybe something like: "To change the functionality of the suffix icon, provide a custom `onSuffixTap` callback and specify an intuitive `suffixIcon`.